tests/tests/test_optfileparser.py

```diff
@@ -147,3 +148,45 @@ def test_required():
 
     of = parse_file("required.opt", "linux,ada,c")
     assert not of.is_dead
+
+
+def run_opt_parser_script(filename, tags=None):
+    """Call e3-opt-parser and returning the corresponding e3.os.process.Run object."""
+    parser_cmd = [
+        "e3-opt-parser",
+        os.path.join(os.path.dirname(__file__), "optfiles", filename),
+    ]
+    if tags is not None:
+        parser_cmd.extend(tags)
+    return Run(parser_cmd)
+
+
+def test_main():
+    """Test the function called by the command-line wrapper to OptFileParse."""
+    p = run_opt_parser_script("tags.opt", None)
+    assert p.status == 0, f"status: {p.status}\n{p.out}"
```

Comment:
`py.test` is able to show the value of `p.status` if the assertion fails, so I would suggest removing the message part (here and below): they break brevity and bring no value

Reply:
Wondering how this is working, but this is brilliant. New patch is up...